test-common/src/main/java/org/apache/kafka/common/test/PreboundSocketFactoryManager.java

```diff
@@ -49,7 +49,16 @@ public ServerSocketChannel openServerSocket(
             ServerSocketChannel socketChannel = getSocketForListenerAndMarkAsUsed(
                 nodeId,
                 listenerName);
-            if (socketChannel != null) {
+            if (socketChannel != null && socketChannel.isOpen()) {
+                return socketChannel;
+            } else if (socketChannel != null) {
+                // bind the server socket with same port
+                socketAddress = new InetSocketAddress(socketAddress.getHostString(), socketChannel.socket().getLocalPort());
+                socketChannel = ServerSocketFactory.INSTANCE.openServerSocket(
+                        listenerName,
+                        socketAddress,
+                        listenBacklogSize,
+                        recvBufferSize);
                 return socketChannel;
             }
```

Comment:
Please combine the null checks. 
Instead of checking for `socketChannel != null` twice, we can combine them into a single block that handles the non-null case.